libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -2166,6 +2176,26 @@ class MapboxNavigation @VisibleForTesting internal constructor(
         registerRoutesPreviewObserver(routesCacheClearer)
     }
 
+    private suspend fun prepareNavigationForRoutesParsing() {
+        withContext(Dispatchers.Main.immediate) {
```

Comment:
NIT: Let's only do this when we current have more than 1 route. Otherwise it doesn't make sense. And though I don't see any reasons why it would break anything, who knows.

Reply:
done